routes/proxy.js

```diff
@@ -258,6 +258,9 @@ module.exports = function (jwtMiddleware) {
      */
     router.all('/:prefix*', function (req, res, next) {
         var prefix = req.params.prefix;
+        var hostHeader = req.protocol + '://' + req.get('host') + req.originalUrl;
+        hostHeader = hostHeader.slice(-1) === '/' ? hostHeader : hostHeader + '/';
+        req.headers['x-forwarded-host'] = hostHeader;
```

Comment:
Furthermore I think this header should only be set when we know the request is going to be proxied (Authorization and Authentication layers are successful). 

This can be done upper this file (Line 30) when we set the 'x-gleaner-user' header as well in the methos `proxy.on('proxyReq', function (proxyReq, req, res, options) {...`